sourcegraph/sourcegraph
# Add "getting started/quick start" checklist tour

Closes https://github.com/sourcegraph/sourcegraph/issues/28322, closes https://github.com/sourcegraph/sourcegraph/issues/32712, closes https://github.com/sourcegraph/sourcegraph/issues/32713, closes https://github.com/sourcegraph/sourcegraph/issues/32714

## Description
This PR:
- Migrates existing "getting started tour for visitors" to a new design
- Refactors to support multiple dynamic content
- Adds new "getting started/quick start" checklist tour for authenticated cloud users

## Test plan
- `SOURCEGRAPHDOTCOM_MODE=true SOURCEGRAPH_API_URL=https://sourcegraph.com sg start web-standalone`
1. Check visitors' (non-authenticated) version
    - Open https://sourcegraph.test:3443/search
    - Go through the "quick start" tour
    - Make sure that all steps work as expected, event logs, marked as completed, as well as closing the tour
2. Check authenticated (non-authenticated) version
   - https://sourcegraph.test:3443/sign-in and sign in
    - https://sourcegraph.test:3443/search?feature-flag-key=quick-start-tour-for-authenticated-users&feature-flag-value=true overrides feature flag
    - Go through the "quick start" tour
    - Make sure that all steps work as expected, event logs, marked as completed, as well as closing the tour
    - https://sourcegraph.test:3443/search?feature-flag-key=quick-start-tour-for-authenticated-users removed feature flag override
4. On-prem (not cloud) mode
   - `SOURCEGRAPHDOTCOM_MODE=false SOURCEGRAPH_API_URL=https://sourcegraph.com sg start web-standalone`
   - Check that no "quick start" tour is shown
> **NOTE:** If you want to reset, delete `quick-start-tour` local storage variable

## Screenshots

<details>

| Visitors tour | Authenticated tour |
| --: | :-- |
| ![image](https://user-images.githubusercontent.com/6717049/159925132-9e555b12-8ed3-40d8-bb62-082b4565aeb9.png) | ![image](https://user-images.githubusercontent.com/6717049/159936515-d274c1e9-e834-48f3-9f0e-3114aeb89ed0.png) |
| ![image](https://user-images.githubusercontent.com/6717049/159919027-93278929-50f6-4cf7-98f7-aceab2fced8b.png) | ![image](https://user-images.githubusercontent.com/6717049/159936610-4922c1b2-cc34-4511-ab56-7024c0445ad7.png)|
| ![image](https://user-images.githubusercontent.com/6717049/159919183-68265528-c959-4bea-a83a-83dfa2bc2a11.png) |  ![image](https://user-images.githubusercontent.com/6717049/159924774-e0baad8c-b1df-4a44-afc3-a46927c7477d.png) |

</details>



## App preview:
- [Link](https://sg-web-erzhtor-add-dynamic-quick-start.onrender.com)
